doc/source/ray-contribute/docs.ipynb

```diff
@@ -156,6 +156,17 @@
     "For example, in the above `autofunction` call, to change the API reference for `ray.tune.integration.docker.DockerSyncer`,\n",
     "you would have to [change the following source file](https://github.com/ray-project/ray/blob/7f1bacc7dc9caf6d0ec042e39499bbf1d9a7d065/python/ray/tune/integration/docker.py#L15-L38).\n",
     "\n",
+    "To show the usage of APIs, it is important to have small usage examples embedded in the API documentation. These should be self contained and run out of the box so a user can copy and paste them into a Python interpreter and play around with them (e.g. if applicable they should point to example data). Users often rely on these examples to build their applications. You can use the Sphinx `testcode` primitive to embed\n",
+    "such examples into the `Examples:` section of the docstrings. For an example look [here](https://github.com/ray-project/ray/blob/5e61fb51400bc712449e85a7476fa4fca80f3b41/python/ray/train/torch/torch_predictor.py#L173-L221)\n",
+    "\n",
+    "These code snippets will be tested in the CI to make sure they will keep working going forward. You can use the `testoutput` primitive to specify the expected output of the code snippet and the CI will check the output and give an error if they don't match. To run the doctests locally, run\n",
+    "\n",
+    "```shell\n",
+    "RAY_MOCK_MODULES=0 make doctest\n",
```

Comment:
"export RAY_MOCK_MODULES=0; make doctest\n"

Reply:
I think this part is better to leave as it is. If people do the `export` and then later run the unit tests (not the doctest), the unit testing behavior will be different and might cause failures (since the environment variable will persist). Without the export, the environment variable will only be defined for the duration of `make doctest`, which is safer :)